code/modules/mechs/equipment/combat.dm

```diff
@@ -802,6 +805,221 @@
 			cocked = FALSE
 			// not being able to fire removes the CH(done in reloadGun now)
 
+/obj/item/mech_equipment/mounted_system/mace
+	name = "\improper NT \"Warhead\" mace"
+	desc = "An exosuit-mounted mace. Doubles as a flail. Handle with care."
+	icon_state = "mech_mace"
+	holding_type = /obj/item/tool/hammer/mace/mech
+	restricted_hardpoints = list(HARDPOINT_LEFT_HAND, HARDPOINT_RIGHT_HAND)
+	matter = list(MATERIAL_PLASTEEL = 15, MATERIAL_STEEL = 10)
+	origin_tech = list(TECH_COMBAT = 4, TECH_MAGNET = 3)
+	/// Determines what mode our mace is. FALSE is mace, TRUE is flail
+	var/flail_mode = FALSE
+
+/obj/item/mech_equipment/mounted_system/mace/Initialize()
+	. = ..()
+	var/obj/item/tool/hammer/mace/mech/holdin = holding
+	holdin.wielded = TRUE
+
+/obj/item/mech_equipment/mounted_system/mace/attack_self(mob/user)
+	. = ..()
+	if(.)
+		var/obj/item/tool/hammer/mace/mech/holdin = holding
+		holdin.update_flail_mode(flail_mode)
+		flail_mode = !flail_mode
+		owner.update_icon()
+
+/obj/item/mech_equipment/mounted_system/mace/resolve_attackby(mob/living/target, mob/user, params)
+	if(ishuman(target))
+		var/mob/living/carbon/human/targ = target
+		if(targ.stats.getStat(STAT_VIG) > STAT_LEVEL_EXPERT)
```

Comment:
Flat stat check? Shouldn't there be a Prob()?

Reply:
Probn't

Reply:
I would *usually* agree with it, but we should avoid any kind of RNG in PvP (it is PvP). This is fine currently, or the damage can just start to decrease after certain VIG (so if godlike is 80 (iirc?), as example at 40 VIG it starts to decrease, at 60 VIG damage is halved, at 79 it, like 2 brute; *The mace barely grazes you!*)

Reply:
Making the weapon have no effect though is definitely not fine.

Reply:
Rewrote this proc to use an intensity variable instead of a flat skip. Current formula is `STAT_LEVEL_ADEPT / targ.stats.getStat(STAT_VIG)`, meaning expert gives you 0.625x damage,  professional gives you 0.416x damage, and godlike gives you 0.3125x damage. If that feels too light at the higher-tier levels, I could mess with the formula a bit.